src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    if (this.length === 0) {
+      return result;
+    }
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (this[i] !== undefined && this[i] !== null) {
+        result += `${this[i]}${separator}`;
+      } else {
+        result += separator;
+      }
+    }
+
+    if (this[this.length - 1] !== undefined && this[this.length - 1] !== null) {
+      result += this[this.length - 1];
+    }
```

Comment:
```suggestion
      if (i < this.length - 1) {
        result += separator;
      }
```

Reply:
I'm sorry, can you please explain me how to apply this part of the code? I don't have variable 'i' (except for index in the loop before) and I don't understand how it can help me to check the last element of the array.

Reply:
You need do this check inside loop. If you have more questions, ask them in fe_chart

Reply:
Thank you for the hint, done)